### DISCUSSION TITLE
Set default locale using runtimeConfig

### DISCUSSION THREAD

**Discussion by bitbytebit1** [nuxt-modules/i18n#2057] [Ideas]
### Describe the feature

I need to be able to set the defaultLocale using environment variables that will change at run time, would it be possible to add this to the [runtimeConfig](https://v8.i18n.nuxtjs.org/options/runtime-config)?

### Additional information

- [ ] Could this feature be implemented as a module?
- [X] Would you be willing to help implement this feature?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution) (The contribution guideline of nuxt-modules/i18n is compliant with Nuxt too).
- [X] Check existing [discussions](https://github.com/nuxt-modules/i18n/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by kazupon** [comment]
I've just converted to github discussion. 
That idea should be more discussed as feature request.

Please give us more your explain with actually use-case situation.

**Comment by bitbytebit1** [comment]
We have a project that is deployed to 7 different domains that is available in 7 languages, each website needs to have a different default locale, currently we set which domain is being deployed via environment variables, this allows use to deploy the same project to all domains without rebuilding the project by passing in different environment variables.

My current understanding is the only way to set the default language would be at build time using environment variables, is this the recommended approach?

**Comment by randyhoulahan** [comment]
I have a need for a dynamic defaultLocale.   My office is a the UN Convention on Biological Diversity.  Our project nuxtjs 3 frontend for up to 196 drupal back ends (all countries that are parties to our convention).   I did not want to build a separate front end for each but one that serves them all.  My only problem is the dynamic defaultLocale.   We already started dev but this is a blocker.  Not sure what the resolution will be yet.  PR here  or ... separate builds.
